packages/playground/src/components/Toolbar/index.ts

```diff
@@ -0,0 +1,120 @@
+import type { InlineTool, InlineToolAdapter } from '@editorjs/dom-adapters';
+import { type EditorJSModel, type TextRange, Index } from '@editorjs/model';
+import { EventType } from '@editorjs/model';
+import type { Nominal } from '@editorjs/model/dist/utils/Nominal';
+import { ref } from 'vue';
+
+/**
+ * Class determines, when inline toolbar should be rendered
+ * Handles caret selection changes
+ * Forms data required in certain inline tool
+ */
+export class InlineToolbar {
+  /**
+   * Editor model instance
+   * Used for interactions with stored data
+   */
+  #model: EditorJSModel;
+
+  /**
+   * Inline tool adapter instance
+   * Used for inline tools attaching and format apply
+   */
+  #inlineToolAdapter: InlineToolAdapter;
+
+  /**
+   * Current selection range
+   */
+  #selectionRange: TextRange | undefined = undefined;
+
+  /**
+   * Tools that would be attached to the adapter
+   */
+  #tools: InlineTool[];
+
+  /**
+   * Toolbar show state
+   */
+  public show = ref<boolean>(false);
+
+  /**
+   * @class
+   * @param model - editor model instance
+   * @param inlineToolAdapter - inline tool adapter instance
+   * @param tools - tools, that should be attached to adapter
+   */
+  constructor(model: EditorJSModel, inlineToolAdapter: InlineToolAdapter, tools: InlineTool[]) {
+    this.#model = model;
+    this.#inlineToolAdapter = inlineToolAdapter;
+    this.#tools = tools;
+
+    this.#attachTools();
+
+    this.#handleSelectionChange();
+  }
+
+  /**
+   * Handle changes of the caret selection
+   */
+  #handleSelectionChange(): void {
+    /**
+     * Listen to selection change ivents in model
+     */
+    this.#model.addEventListener(EventType.CaretManagerUpdated, (event) => {
+      const selection = window.getSelection();
+
+      /**
+       * Get current input with selection
+       */
+      if (selection) {
+        /**
+         * Do not render inline toolbar for not contenteditable elements
+         */
+        if (selection.focusNode?.nodeType !== Node.TEXT_NODE) {
+          return;
+        }
+      }
```

Comment:
Would be enough to check if caret has textRange. If it has it — that means caret is in the text input

Reply:
as i can see, caret also has text range for native inputs, but i want to check that current node is text and contenteditable